flow-decomposition/src/test/java/com/powsybl/flow_decomposition/NetPositionTest.java

```diff
@@ -122,4 +124,50 @@ void testHvdcLinesNaN() {
         network.getHvdcLine("hvdc_line_FR_1_DE").getConverterStation2().getTerminal().setP(Double.NaN);
         assertNetPositionForHvdc(network, 200.0);
     }
+
+    @Test
+    void testTieLine() {
+        String networkFileName = "NETWORK_SINGLE_LOAD_TWO_GENERATORS_WITH_XNODE.uct";
+
+        Network network = importNetwork(networkFileName);
+        TieLine tieLine = network.getTieLine("FGEN1 11 X     11 1 + X     11 BLOAD 11 1");
+        tieLine.getDanglingLine1().setB(1E-3);
+        tieLine.getDanglingLine2().setB(1E-3);
+
+        LoadFlow.run(network, new LoadFlowParameters().setDc(false));
+        Map<Country, Double> netPositions = NetPositionComputer.computeNetPositions(network);
+
+        double sumAllNetPositions = netPositions.values().stream().mapToDouble(Double::doubleValue).sum();
+        assertEquals(0.0, sumAllNetPositions, DOUBLE_TOLERANCE);
+        assertEquals(-100.17064, netPositions.get(Country.BE), DOUBLE_TOLERANCE);
+        assertEquals(100.17064, netPositions.get(Country.FR), DOUBLE_TOLERANCE);
+    }
+
+    @Test
+    void testTieLineWithDisconnectedDanglingLine() {
+        String networkFileName = "19700101_0000_FO4_UX1.uct";
+        Network network = importNetwork(networkFileName);
+        network.getDanglingLineStream().forEach(danglingLine -> {
+            danglingLine.setR(1E-1);
+            danglingLine.setB(1E-3);
+        });
+
+        LoadFlow.run(network, new LoadFlowParameters().setDc(false));
+        Map<Country, Double> netPositions = NetPositionComputer.computeNetPositions(network);
+        double sumAllNetPositions = netPositions.values().stream().mapToDouble(Double::doubleValue).sum();
+        double sumAllUnpairedDanglingLines = network.getDanglingLineStream().filter(danglingLine -> !danglingLine.isPaired())
+                .mapToDouble(danglingLine -> danglingLine.getTerminal().isConnected() && !Double.isNaN(danglingLine.getTerminal().getP()) ? danglingLine.getTerminal().getP() : 0.0)
+                .sum();
+        assertEquals(sumAllUnpairedDanglingLines, sumAllNetPositions, DOUBLE_TOLERANCE);
+
+        // disconnect dangling line 1 of a tie line
+        // => dangling line 2's contribution to net position must affect sum of all net positions
+        TieLine tieLine = network.getTieLine("XBF00011 BF000011 1 + XBF00011 FB000011 1");
+        tieLine.getDanglingLine1().getTerminal().disconnect();
+        LoadFlow.run(network, new LoadFlowParameters().setDc(false));
+        netPositions = NetPositionComputer.computeNetPositions(network);
+        sumAllNetPositions = netPositions.values().stream().mapToDouble(Double::doubleValue).sum();
+        double deltaNetPosition = tieLine.getDanglingLine2().getTerminal().getP();
+        assertEquals(sumAllUnpairedDanglingLines + deltaNetPosition, sumAllNetPositions, DOUBLE_TOLERANCE);
```

Comment:
```suggestion
        TieLine tieLine = network.getTieLine("XBF00011 BF000011 1 + XBF00011 FB000011 1");
        assertEquals(-779.848, tieLine.getTerminal1().getP(), DOUBLE_TOLERANCE);
        assertEquals(780.609, tieLine.getTerminal2().getP(), DOUBLE_TOLERANCE);
        Map<Country, Double> netPositions = NetPositionComputer.computeNetPositions(network);
        assertEquals(-800.626, netPositions.get(Country.BE), DOUBLE_TOLERANCE);
        assertEquals(1402.142, netPositions.get(Country.FR), DOUBLE_TOLERANCE);
        assertEquals(-1400.860, netPositions.get(Country.DE), DOUBLE_TOLERANCE);
        assertEquals(-799.345, netPositions.values().stream().mapToDouble(Double::doubleValue).sum(), DOUBLE_TOLERANCE);
        assertEquals(-799.345, network.getDanglingLineStream(DanglingLineFilter.UNPAIRED).mapToDouble(danglingLine -> danglingLine.getTerminal().getP()).sum(), DOUBLE_TOLERANCE);

        tieLine.getTerminal1().disconnect();

        LoadFlow.run(network, new LoadFlowParameters().setDc(false));
        assertEquals(0, tieLine.getTerminal1().getP(), DOUBLE_TOLERANCE);
        assertEquals(0.032, tieLine.getTerminal2().getP(), DOUBLE_TOLERANCE);
        Map<Country, Double> newNetPositions = NetPositionComputer.computeNetPositions(network);
        assertEquals(-800.774, newNetPositions.get(Country.BE), DOUBLE_TOLERANCE);
        assertEquals(1402.930, newNetPositions.get(Country.FR), DOUBLE_TOLERANCE);
        assertEquals(-1401.501, newNetPositions.get(Country.DE), DOUBLE_TOLERANCE);
        assertEquals(-799.345, newNetPositions.values().stream().mapToDouble(Double::doubleValue).sum(), DOUBLE_TOLERANCE);
        assertEquals(-799.345, network.getDanglingLineStream(DanglingLineFilter.UNPAIRED).mapToDouble(danglingLine -> danglingLine.getTerminal().getP()).sum(), DOUBLE_TOLERANCE);
```
I do not agree to your statement "dangling line 2's contribution to net position must affect sum of all net positions".
As you run an AC load flow, net position will not change much provided that nothing is saturated.

Please add hard coded expected values. It does not cost much, but it really helps to understand the code and find errors later on.

Could you add this test please ?
```
    @Test
    void testLinesSingleSideDisconnected() {
        Network network = Network.read("testCase.xiidm", getClass().getResourceAsStream("testCase.xiidm"));
        network.getBranch("NNL2AA1  BBE3AA1  1").getTerminal1().disconnect();
        LoadFlow.run(network, new LoadFlowParameters().setDc(true));
        assertNetPosition(network, 1500.0, 0.0, -2500.0);
    }
```

Reply:
This test only makes sense if the treatment is different for paired dangling lines with one of them being disconnected.
I removed the test and added yours.